lib/spack/spack/fetch_strategy.py

```diff
@@ -421,6 +430,79 @@ def fetch(self):
         tty.msg('Using cached archive: %s' % path)
 
 
+class PyPiFetchStrategy(URLFetchStrategy):
+    """Fetches packages directly from PyPi using XML-RPC to query the API."""
+
+    enabled = True
+    required_attributes = ['pypi']
+
+    def __init__(self, pypi=None, digest=None, **kwargs):
+        # If pypi or digest are provided in the kwargs, then prefer
+        # those values.
+        self.pypi = kwargs.get('pypi', None)
+        if not self.pypi:
+            self.pypi = pypi
+
+        self.version = kwargs.get('version', None)
+
+        self.digest = kwargs.get('md5', None)
+        if not self.digest:
+            self.digest = digest
+
+        self.expand_archive = kwargs.get('expand', True)
+        self.extra_curl_options = kwargs.get('curl_options', [])
+        self._curl = None
+
+        self.extension = kwargs.get('extension', None)
+
+        if not self.pypi:
+            raise ValueError("PyPiFetchStrategy requires the name of the "
+                             "package on PyPi for fetching.")
```

Comment:
Oh, this line explains what the variable `self.pypi` is (that will not be a problem once arg comments are put in the docstring).  But maybe it should be called `self.pypi_package_name` (and name the kwarg the same, especially if it is not user-facing).